src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/XmlSerializerGenerator.cs

```diff
@@ -124,21 +122,94 @@ private List<Type> CollectXmlSerializerTypes(Assembly assembly, List<XmlMapping>
+                }
+
+                var operations = operationsProperty.GetValue(contract);
+                Type operationsType = operations.GetType();
+                MethodInfo getCount = operationsType.GetMethod("get_Count");
+                int count = (int)getCount.Invoke(operations, new object[] { });
+                for (int i = 0; i < count; ++i)
                 {
-                    XmlSerializerOperationBehavior behavior = operation.Behaviors.Find<XmlSerializerOperationBehavior>();
+                    MethodInfo getItem = operationsType.GetMethod("get_Item");
+                    var operation = getItem.Invoke(operations, new object[] { i });
+                    if(operation == null)
+                    {
+                        throw new ToolRuntimeException("operation is null");
+                    }
+
+                    var getBehaviors = operation.GetType().GetMethod("get_Behaviors");
+                    if (getBehaviors == null)
+                    {
+                        ToolConsole.WriteError($"Not found method get_Behaviors in type {operation.GetType()}");
+                        throw new ToolRuntimeException();
+                    }
+
+                    var behaviors = getBehaviors.Invoke(operation, new object[] { });
+                    var find = behaviors.GetType().GetMethod("Find");
+                    if (find == null)
+                    {
+                        ToolConsole.WriteError($"Not found method Find in type {behaviors.GetType()}");
+                        throw new ToolRuntimeException();
+                    }
+
+                    Type XmlSerializerOperationBehaviorType = Tool.SMAssembly.GetType("System.ServiceModel.Description.XmlSerializerOperationBehavior");
+                    if (XmlSerializerOperationBehaviorType == null)
+                    {
+                        ToolConsole.WriteError($"Not found type System.ServiceModel.Description.XmlSerializerOperationBehaviorType in {Tool.SMAssembly.FullName}");
+                        throw new ToolRuntimeException();
+                    }
+
+                    MethodInfo FindXmlSerializerOperationBehavior = find.MakeGenericMethod(new Type[] { XmlSerializerOperationBehaviorType });
+                    if(FindXmlSerializerOperationBehavior == null)
+                    {
+                        ToolConsole.WriteError($"Not found method Find<XmlSerializerOperationBehavior> in operation.Behaviors");
+                        throw new ToolRuntimeException();
+                    }
+
+                    var behavior = FindXmlSerializerOperationBehavior.Invoke(behaviors, new object[] { });
                     if (behavior != null)
                     {
-                        foreach (XmlMapping map in behavior.GetXmlMappings())
+                        MethodInfo getXmlMappings = behavior.GetType().GetMethod("GetXmlMappings");
+                        if(getXmlMappings == null)
```

Comment:
The type of behavior will be System.ServiceModel.Description.XmlSerializerOperationBehavior which you got above. You can move this out of the foreach loop including getting the GetXmlMappings method.